stravalib/model.py

```diff
@@ -149,58 +188,132 @@ def naive_datetime(value: Optional[AllDateTypes]) -> Optional[datetime]:
 class DeprecatedSerializableMixin(BaseModel):
     """
     Provides backward compatibility with legacy BaseEntity
+
+    Inherits from the `pydantic.BaseModel` class.
     """
 
+    # TODO: i used the output type from this class
+    # could dict also be an int or float??
     @classmethod
-    def deserialize(cls, attribute_value_mapping: Dict):
+    def deserialize(
+        cls, attribute_value_mapping: dict[str, str]
+    ) -> DeprecatedSerializableMixin:
         """
         Creates and returns a new object based on serialized (dict) struct.
+
+        Parameters
+        ----------
+        attribute_value_mapping : dict
+            A dictionary representing the serialized data.
+
+        Returns
+        -------
+        DeprecatedSerializableMixin
+            A new instance of the class created from the serialized data.
+
+        Deprecated
+        ----------
+        1.0.0
+            The `deserialize()` method is deprecated in favor of `parse_obj()` method.
+            For more details, refer to the Pydantic documentation:
+            https://docs.pydantic.dev/usage/models/#helper-functions
+
+
         """
-        warn_method_deprecation(
+        exc.warn_method_deprecation(
             cls,
             "deserialize()",
             "parse_obj()",
             "https://docs.pydantic.dev/usage/models/#helper-functions",
         )
         return cls.parse_obj(attribute_value_mapping)
 
-    def from_dict(self, attribute_value_mapping: Dict):
+    # TODO: Same question here about dict elements (strings or str, Union[str,int])
+    def from_dict(self, attribute_value_mapping: dict[str, str]) -> None:
         """
-        Deserializes v into self, resetting and ond/or overwriting existing fields
+        Deserializes v into self, resetting and ond/or overwriting existing
+        fields
+
+        Parameters
+        ----------
+        attribute_value_mapping : dict
+            A dictionary that will be deserialized into the parent object.
+
+        Deprecated
+        ----------
+        1.0.0
+            The `from_dict()` method is deprecated in favor of `parse_obj()` method.
+            For more details, refer to the Pydantic documentation:
+            https://docs.pydantic.dev/usage/models/#helper-functions
         """
-        warn_method_deprecation(
+        # TODO: Argument 1 to "warn_method_deprecation" has incompatible type "str"; expected "type
+        exc.warn_method_deprecation(
             self.__class__.__name__,
```

Comment:
The function except the class itself as the first argument, not its name.
```suggestion
            self.__class__
```